### PR TITLE
refactor: Modernized/optimized general code, bug fixing and more

### PR DESCRIPTION
**Pull request by karifeld**
refactor: Utilize self globals and gui::components in most code
fix(Vehicle Preview): addresses the issues mentioned in #119

fix(Infinite Clip): Fixed an issue where disabling it doesnt work
fix(No Ragdoll): Fixed an issue where disabling it doesnt work, removed unnecessary calls to natives as they are hardcoded to false for local ped.
fix(Spawn menu): Fixed placement of `ImGui::ListBoxFooter();` (It caused an issue where if `vehicles.json` was not present, The ListBox would be the wrong size)

fix: `self_gloabls.cpp` name typo

refactor(Mobile): Only call `TASK::CLEAR_PED_TASKS_IMMEDIATELY();` if pv_teleport_into global is true and player ped is in a vehicle
feat(Weapons): Remove current weapon

refactor: Added missing variable in calls to self globals
refactor: Utilize usage of ControllerInputs

### CODE REVIEW

**Comment by xM4ddy** [comment]
Hey looking through the changes and wondering about your decision to do something like `Vehicle vehicle = self::veh;` rather than using `self::veh` throughout. (Not saying it's an issue just looking to clarify something.)

**Comment by karifeld** [comment]
Well, its an optimization habit that ive carried over outside of Lua.

in Lua, if you were to run the following:
`for i = 1, 1000000 do 
   local x = math.sin(i) 
end`

`local sin = math.sin 
for i = 1, 1000000 do
    local x = sin(i) 
end`

The latter would run faster than the former. because Lua's main data structure is a table. if you were to a global function instead of a local, it will be slower because globals are stored in a table and are accessed via hash lookup. On the other hand, If you declare a local variable outside of a loop, the resulting code would be faster as they are accessed by index.

Now, im not expert but i dont think declaring a local variable really makes a difference in CPP.
which is why theres another reason, Readability and tidiness.

Compare both codes. one has a location variable declared and one does not.
	
	Vector3 vector;
	vector.x = self::pos.x * 2;
	vector.y = self::pos.y * 2;
	vector.z = self::pos.z * 2;
	
.

	Vector3 location = self::pos;
	
	Vector3 vector;
	vector.x = location.x * 2;
	vector.y = location.y * 2;
	vector.z = location.z * 2;
	
The latter looks neater and much more comprehensive than the former, Consistency is the most important aspect of a coding style. I think it should be best practiced even in other projects.

File: BigBaseV2/src/backend/looped/self/no_ragdoll.cpp [hunk 1]
```diff
@@ -5,13 +5,6 @@ namespace big
...
 	void looped::self_no_ragdoll()
 	{
-		Ped player = PLAYER::PLAYER_PED_ID();
-
-		if (g->self.no_ragdoll) {
-			PED::SET_PED_CAN_RAGDOLL(player, !g->self.no_ragdoll);
-			PED::SET_PED_CAN_RAGDOLL_FROM_PLAYER_IMPACT(player, !g->self.no_ragdoll);
-			PED::SET_PED_RAGDOLL_ON_COLLISION(player, !g->self.no_ragdoll);
-		}
-
```
Comment (Yimura) [inline BigBaseV2/src/backend/looped/self/no_ragdoll.cpp]: You sure you want to always run this native, you're even executing the native when the feature is disabled, possibly causing side effects in missions. Have you properly tested this?

File: BigBaseV2/src/backend/looped/weapons/infinite_mag.cpp [hunk 2]
```diff
@@ -5,8 +5,6 @@ namespace big
 {
 	void looped::weapons_infinite_mag()
 	{
-		if (g->weapons.infinite_mag) {
-			WEAPON::SET_PED_INFINITE_AMMO_CLIP(PLAYER::PLAYER_PED_ID(), g->weapons.infinite_mag);
-		}
```
Comment (Yimura) [inline BigBaseV2/src/backend/looped/weapons/infinite_mag.cpp]: Forcefully running this native each tick even when disabled might cause side effects.

File: BigBaseV2/src/backend/looped/vehicle/fly.cpp [hunk 3]
```diff
@@ -1,6 +1,7 @@
 #include "backend/looped/looped.hpp"
 #include "natives.hpp"
 #include "util/entity.hpp"
+#include "gta/enums.hpp"
```
Comment (Yimura) [inline BigBaseV2/src/backend/looped/vehicle/fly.cpp]: Sort includes alphabetically

**Review by Yimura — CHANGES_REQUESTED** [review]
In general all looks good, if you resolve the few comments I have here and there then we can merge it into master.

Also CI is failing, fix the error the below error:
`src\backend\looped\vehicle\fly.cpp(14): error C4700: uninitialized local variable 'ped' used`

**Comment by maybegreat48** [comment]
Infinite ammo (not infinite clip) can be simplified to ```WEAPON::SET_PED_INFINITE_AMMO(self::ped, g.weapons.infinite_ammo, NULL);```

**Comment by karifeld** [comment]
@Yimura Changes have been made, let me know what you think.

@maybegreat48  Thanks for the contribution.

**Review by Yimura — APPROVED** [review]

**Comment by Yimura** [comment]
If there are no more changes to be added then it's ready to be merged into master.

**Comment by karifeld** [comment]
That should be it, you may merge at anytime

**Comment by Yimura** [comment]
Alright @xM4ddy will take one final look at it and then it'll be merged.

**Comment by xM4ddy** [comment]
> Infinite ammo (not infinite clip) can be simplified to `WEAPON::SET_PED_INFINITE_AMMO(self::ped, g.weapons.infinite_ammo, NULL);`

I'm trying to remember why I did that. It was actually a conscious choice actually. I may have just been some weird preference for the behavior of doing it the way I did.

**Comment by Yimura** [comment]
There might be some odd scenario where the game resets the toggle.

**Comment by xM4ddy** [comment]
Both seem to work so if I remember what it was and its important I'll just make another commit.

**Pull request merged**